mpyg321/mpyg321.py

```diff
@@ -63,20 +32,86 @@ class MPyg321PlayerSeekError(MPyg321PlayerError):
     pass
 
 
-mpg_errors = {
-        "Error opening stream:": MPyg321PlayerFileError,
-        "failed to parse given eq file:": MPyg321PlayerFileError,
-        "Corrupted file:": MPyg321PlayerFileError,
-        "Unknown command:": MPyg321PlayerCommandError,
-        "Unfinished command:": MPyg321PlayerCommandError,
-        "Unknown command or no arguments:": MPyg321PlayerArgumentError,
-        "invalid arguments for": MPyg321PlayerArgumentError,
-        "Missing argument to": MPyg321PlayerArgumentError,
-        "failed to set eq:": MPyg321PlayerEQError,
-        "Error while seeking": MPyg321PlayerSeekError,
-        "empty list name": MPyg321PlayerError,
-        "No track loaded!": MPyg321PlayerError
-}
+mpgouts = [
```

Comment:
Thank you! I think there was a misunderstanding: I was expecting to have all the constants first, and then all the classes. Now we have: the error classes->the constants->MPyg321 class
We should have: the constants->the error classes->Mpyg321 class

Reply:
Sorry for the misunderstanding :/

Reply:
It's fine, I misinterpreted it. By constants do you mean `mpgerrors`, `mpgouts` and `mpgcodes`?

Reply:
Unfortunately `mpgerrors` can't be put before the error classes because it contains those classes as part of the dictionaries. Alternatively I can rewrite the dictionaries to have an action instead of an exception and then raise the relevant exception instead of dealing with the exception instantly (similar to the `mpgouts` list of dictionaries). I think I'll do that instead.

Reply:
Good point! What about putting all the error related stuff (classes and mpgerrors) in a separate class for clarity?

Reply:
I've basically rewritten the `mpg_errors` list to the following structure:

```
[
   {
        "message": "empty list name",
        "action": "generic_error"
   },
   {...},
   ...
]
```

And then, in `handle_error`:

```
for mpg_error in mpg_errors:
    if error["message"] in output:
        action = error["action"]
        if action == "generic_error":
            raise MPyg321PlayerError
       ...
```

Which is consistent with `process_output`. Hopefully that's okay!